### DISCUSSION TITLE
ControlNet and Inpaint problem

### DISCUSSION THREAD

**Discussion by CrazyMaxTM** [Mikubill/sd-webui-controlnet#1888] [Q&A]
In the "Inpaint" mode "Only masked" if the "Mask blur" parameter is greater than zero, ControlNet returns an enlarged tile
![BAG ControlNet_01](https://github.com/Mikubill/sd-webui-controlnet/assets/121190531/f33502bf-5fe5-4263-9030-20521c2cfbaf)
![BAG ControlNet_03](https://github.com/Mikubill/sd-webui-controlnet/assets/121190531/a48adc95-04df-474b-8793-b6131ba88f67)
![BAG ControlNet_04](https://github.com/Mikubill/sd-webui-controlnet/assets/121190531/d2818a14-6baa-4d1c-b4af-723426033272)
![BAG ControlNet_05](https://github.com/Mikubill/sd-webui-controlnet/assets/121190531/d790bb90-9ba0-4dfe-b66f-5f753330611d)
If the "Mask blur" parameter is equal to zero, then the size of the tile corresponds to the original
![BAG ControlNet_02](https://github.com/Mikubill/sd-webui-controlnet/assets/121190531/59a4d9bf-832a-4863-96b0-9b616440d02a)
Changing "Resize Mode" do not help to avoid this problem

**Comment by DarkStoorM** [comment]
This is exactly the issue that is holding me from updating to newer webui versions. I'm currently forced to stay on **1.3.0**(!) webui version, because at some point, inpainting with Tile Resample became completely unusable due to the incorrect size of the result image in Only Masked input - it's enlarged by the mask blur value for some reason. I can't even find which commit broke this.

Masked image does not fit the input anymore and changing the blur to 0 makes no sense, because the image won't blend in, resulting in a visible seam between the mask and the input. In my old version, the inpainted area is **perfectly** downscaled back.

I actually thought it was a ControlNet issue, but it was not. I moved my working CNet v1.1.232 to a separate webui with v1.5.2 and I'm still facing the same issue. It's been broken like that for a long time...

**Comment by lllyasviel** [comment]
For anyone who still interested - this is fixed in latest controlnet (>1.1.311) and webui (>1.5.2)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/0c7bbe99-fccd-41fe-ad00-07d9815b6665)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/cffe9d39-83b0-4cc2-936b-160d803103a3)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/cdca5bce-f1fc-4197-93c5-8145c56b5865)

(only masked)
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/a97936e5-9bef-4651-b9cf-b45aafa03bd2)

**Comment by DarkStoorM** [comment]
> For anyone who still interested - this is fixed in latest controlnet (>1.1.311) and webui (>1.5.2)

I just checked it in the A1111 `1.6.0 pre-release`, it works \o/ I can finally migrate without CNet Tile inpainting breaking things 👍

**Comment by catboxanon** [comment]
Yeah, I should have left a comment here before. https://github.com/Mikubill/sd-webui-controlnet/pull/1907 and https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/12470 resolved this a few weeks back.
